public/src/pages/views/CourseView.tsx

```diff
@@ -44,18 +45,9 @@ export class CourseView extends React.Component<ICourseViewProps, ICourseViewSta
     }
 
     private handleSearch(query: string): void {
-        query = query.toLowerCase();
-        const filteredData: Course[] = [];
-        this.props.courses.forEach((course) => {
-            if (course.getName().toLowerCase().indexOf(query) !== -1
-                || course.getCode().toLowerCase().indexOf(query) !== -1
-                || course.getYear().toString().indexOf(query) !== -1
-            ) {
-                filteredData.push(course);
-            }
-        });
+        const filteredData = searchForCourses(this.props.courses, query);
         this.setState({
-            courses: filteredData,
+            courses: filteredData as Course[],
```

Comment:
Can’t you just call `searchForCourses` directly here.

Reply:
Unfortunately, you can't. It would result in type mismatch: courses in props is `Course[]`, and the method returns `Course[] | Enrollment[]`. TypeScript does not approve.

Reply:
Or actually you can if you assert the type right after the method call. Changed.

Reply:
ok, np